src/snowflake/snowpark/_internal/temp_table_auto_cleaner.py

```diff
@@ -54,53 +46,26 @@ def _delete_ref_count(self, name: str) -> None:
         self.ref_count_map[name] -= 1
         if self.ref_count_map[name] == 0:
             self.ref_count_map.pop(name)
```

Comment:
let's do not do pop here, if we keep the stop method, we can probably sent a teletemetry on close for the temp table that we didn't clean up (ones that whose reference that is 0, and ones that whose reference is not 0)

Reply:
I think we can have a telemetry here about 1) the number of temp tables cleaned up (value is 0) 2) the total number of temp table created by cache_result (length of the dict), when closing the session. So we can understand the percentage of temp table created from cache_result that are cleaned up

Reply:
I think actually it's a good point, and I added such telemetry and corresponding tests